gen/field.go

```diff
@@ -81,57 +86,95 @@ func (f fieldGroupGenerator) DefineStruct(g Generator) error {
 }
 
 func (f fieldGroupGenerator) ToWire(g Generator) error {
-	// TODO(abg): Default values
 	return g.DeclareFromTemplate(
 		`
 		<$wire := import "github.com/thriftrw/thriftrw-go/wire">
 
 		<$v := newVar "v">
-		func (<$v> *<.Name>) ToWire() <$wire>.Value {
+		func (<$v> *<.Name>) ToWire() (<$wire>.Value, error) {
     		<$fields := newVar "fields">
     		<$i := newVar "i">
-			// TODO check if required fields that are reference types are nil
+			<$wVal := newVar "w">
 
-			var <$fields> [<len .Fields>]<$wire>.Field
-			<$i> := 0
+			<if and .IsUnion (len .Fields)>
```

Comment:
actually never mind, I see you're validating `required` before generation, which is fine.

Maybe we should move just the `IsUnion` validation to a separate function that's private?